rpc/lib/client/ws_client.go

```diff
@@ -53,14 +47,18 @@ type WSClient struct {
 	reconnectAfter  chan error            // reconnect requests
 	readRoutineQuit chan struct{}         // a way for readRoutine to close writeRoutine
 
+	// Maximum reconnect attempts (0 or greater; default: 25).
+	maxReconnectAttempts int
+
+	// Support both ws and wss protocols
+	protocol string
+
+	reconnecting bool
+
 	wg sync.WaitGroup
 
 	mtx            sync.RWMutex
 	sentLastPingAt time.Time
-	reconnecting   bool
```

Comment:
note this breaks our convention, where we group fields protected by lock together

```go
mtx sync.Mutex
fieldA typeA
fieldB typeB
```